pkg/asset/machines/master.go

```diff
@@ -606,3 +610,42 @@ func IsMachineManifest(file *asset.File) bool {
 		return matched
 	}
 }
+
+func generateSecretAssets(resources []corev1.Secret, fileName string) (assets []*asset.File, err error) {
+
+	var objects []interface{}
+	for _, r := range resources {
+		objects = append(objects, r)
+	}
+
+	return generateAssets(objects, fileName)
+}
+
+func generateHostAssets(resources []baremetalhost.BareMetalHost, fileName string) (assets []*asset.File, err error) {
+
+	var objects []interface{}
+	for _, r := range resources {
+		objects = append(objects, r)
+	}
+
+	return generateAssets(objects, fileName)
+}
+
+func generateAssets(objects []interface{}, fileName string) (assets []*asset.File, err error) {
```

Comment:
Still looking for this function (and now the other two generateXXXAssets) functions to be renamed since the functions do not generate assets.